os/kernel/binary_manager/binary_manager_recovery.c

```diff
@@ -226,26 +226,24 @@ void binary_manager_deactivate_rtthreads(int bin_idx)
  *	 This function deactivates RT threads and unblocks fault message sender.
  *
  ****************************************************************************/
-void binary_manager_recover_userfault(uint32_t assert_pc)
+void binary_manager_recover_userfault(void)
 {
 	int bin_idx;
-	struct tcb_s *tcb;
 
 #ifdef CONFIG_SUPPORT_COMMON_BINARY
-	if (is_common_library_space((void *)assert_pc)) {
-		/* If a fault happens in common library, it needs to reload all user binaries */
-		int bin_count = binary_manager_get_ucount();
-		for (bin_idx = 1; bin_idx <= bin_count; bin_idx++) {
-			/* Exclude its all children from scheduling if the binary is registered with the binary manager */
-			binary_manager_deactivate_rtthreads(bin_idx);
-		}
-		/* Send fault message and Unblock fault message sender */
-		return binary_manager_unblock_fault_message_sender(BM_CMNLIB_IDX);
-	}
-#endif
+	/* If a fault happens in common binary or user binaries, it needs to reload all user binaries */
+	int bin_count = binary_manager_get_ucount();
 
+	for (bin_idx = 1; bin_idx <= bin_count; bin_idx++) {
```

Comment:
Is there any possibility that rtthread is in common binary?

Reply:
The common binary has no thread list additionally and its threads are linked to a threads list of app binary.
So all threads of common binary are deactivated here by deactivating all user binaries.